src/main/java/appeng/items/parts/ItemMultipartRendering.java

```diff
@@ -0,0 +1,33 @@
+package appeng.items.parts;
+
+
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.item.ItemStack;
+
+import appeng.bootstrap.IItemRendering;
+import appeng.bootstrap.ItemRenderingCustomizer;
+
+
+public class ItemMultipartRendering extends ItemRenderingCustomizer
+{
+
+	private final ItemMultiPart item;
+
+	public ItemMultipartRendering( ItemMultiPart item )
+	{
+		this.item = item;
+	}
+
+	@Override
+	public void customize( IItemRendering rendering )
+	{
+		rendering.meshDefinition( this::getItemMeshDefinition );
+	}
+
+	private ModelResourceLocation getItemMeshDefinition( ItemStack is )
+	{
+		// TODO: Avoid object creation here. This happens every frame
```

Comment:
The object leaves the scope of the method in this case. I know the JVM is very good at optimization but in this case the data flow is complex and goes through several classes outside the scope of this function. 
Besides, it's simply not necessary since the object can already be constructed in the Part Type.